tools/heir-opt.cpp

```diff
@@ -441,6 +441,102 @@ void tosaToBooleanFpgaTfhePipeline(const std::string &yosysFilesPath,
         pm.addPass(createSCCPPass());
       });
 }
+
+struct TosaToJaxiteOptions : public PassPipelineOptions<TosaToJaxiteOptions> {
+  PassOptions::Option<bool> abcFast{*this, "abc-fast",
+                                    llvm::cl::desc("Run abc in fast mode."),
+                                    llvm::cl::init(false)};
+
+  PassOptions::Option<int> unrollFactor{
+      *this, "unroll-factor",
+      llvm::cl::desc("Unroll loops by a given factor before optimizing. A "
+                     "value of zero (default) prevents unrolling."),
+      llvm::cl::init(0)};
+
+  PassOptions::Option<std::string> entryFunction{
+      *this, "entry-function", llvm::cl::desc("Entry function to secretize"),
+      llvm::cl::init("main")};
+};
+
+void tosaToJaxitePipeline(const std::string &yosysFilesPath,
+                          const std::string &abcPath) {
+  PassPipelineRegistration<TosaToJaxiteOptions>(
+      "tosa-to-boolean-jaxite", "Arithmetic modules to jaxite pipeline.",
+      [yosysFilesPath, abcPath](OpPassManager &pm,
+                                const TosaToJaxiteOptions &options) {
+        // Secretize inputs
+        pm.addPass(createSecretize(SecretizeOptions{options.entryFunction}));
+
+        // TOSA to linalg
+        tosaToLinalg(pm);
+
+        // Bufferize
+        oneShotBufferize(pm);
+
+        // Affine
+        pm.addNestedPass<FuncOp>(createConvertLinalgToAffineLoopsPass());
+        pm.addNestedPass<FuncOp>(memref::createExpandStridedMetadataPass());
+        pm.addNestedPass<FuncOp>(affine::createAffineExpandIndexOpsPass());
+        pm.addNestedPass<FuncOp>(memref::createExpandOpsPass());
+        pm.addNestedPass<FuncOp>(affine::createSimplifyAffineStructuresPass());
+        pm.addNestedPass<FuncOp>(affine::createAffineLoopNormalizePass(true));
+        pm.addPass(memref::createFoldMemRefAliasOpsPass());
+        pm.addPass(createExpandCopyPass());
+        pm.addNestedPass<FuncOp>(affine::createAffineLoopNormalizePass(true));
+        pm.addNestedPass<FuncOp>(affine::createLoopFusionPass(
+            0, 0, true, affine::FusionMode::Greedy));
+        pm.addPass(affine::createAffineScalarReplacementPass());
+        pm.addPass(createForwardStoreToLoad());
+
+        // Cleanup
+        pm.addPass(createMemrefGlobalReplacePass());
+        // Commenting out createArithNarrowing since its causing build failures
+        // in OSS for reasons unknown.
+        // FIXME: Uncomment before submission.
```

Comment:
what was the build error?

Reply:
Thanks for help debugging this. 

For posterity: The pass was failing because it was expecting an ArrayRef for vector of bitwidths and failing since its a temporary variable. Declaring a `static std::vector<unsigned> bitWidths = {1, 2, 4, 8, 16};` fixes it